src/Functions/goodness.jl

```diff
@@ -0,0 +1,325 @@
+#= Orthogonal decomposition function to convert F to an invariant basis (K₁, K₂, K₃) for isotropic hyperelastic strain (Criscione JC et al. JMPS, 2000)
+K = FToK123(F) and K = orth_decomp(F) 
+In the current formulation orth_decomp is not autodiff compatible, but FToK123(F) ≈ orth_decomp(F)
+Benchmarking suggests that FToK123 is also about twice as fast as orth_decomp
+K is of dims a Vector{Vector{T}} of dim [1:3][1:nels]
+Outputs: K = [K₁ K₂ K₃] where each term is a vector of length nels =#
+
+function FToK123(F::AbstractMatrix{T}) where {T}
+    @assert size(F) == (3, 3)
+    g = Vector{T}(undef,3)
+      
+    C = transpose(F)*F  
+    λ = sqrt.(eigen(C).values)
+
+    K₁ = log(λ[1]*λ[2]*λ[3])
+    g = log.(λ) .- (K₁/3)
+    K₂ = sqrt(sum(g.^2))
+    K₃ = 3*sqrt(6)*g[1]*g[2]*g[3]/(K₂^3)
+    return [K₁ K₂ K₃]
+end
+
+function FToK123(F::AbstractVector{<:AbstractMatrix{T}}) where {T}
+    nels = length(F)
+    Kel = map(i -> FToK123(F[i]), 1:nels)
+    K = [[Kel[i][j] for i in 1:nels] for j in 1:3]
+    return K
+end
+
+function orth_decomp(F::AbstractMatrix{T}) where {T}
+    @assert size(F) == (3, 3)
+    K = Vector{T}(undef,3)
+    
+    J = det(F)
+    B = F*transpose(F)
+    V = sqrt(B)
+    η = log(V)
+    devη = η - (1/3)*(tr(η))*Matrix(I,3,3)
+
+    K[1] = log(J)
+    K[2] = sqrt(tr(devη^2))
+    Φ = devη/K[2]
+    K[3] = 3*sqrt(6)*det(Φ)
+    return K # warning: final K values may need to be modified to ensure they are within physical limits (e.g., K₃ ∈ [-1 1])
+end
+
+function orth_decomp(F::AbstractVector{<:AbstractMatrix{T}}) where {T}
+    nels = length(F)
+    Kel =  map(i -> orth_decomp(F[i]), 1:nels)
+    K = [[Kel[i][j] for i in 1:nels] for j in 1:3] # transposing the vector of vectors to yield K[1:3][el]
+    return K
+end
+
+#= Function that produces stress and kinematic terms used to construct the virtual fields
+stressTerms, kinTerms = sensitivityFieldFunctions(:ConstitutiveModel) 
+stressTerms = fns[1]
+kinTerms = fns[2]
+Outputs: stressTerms[i,j] (i.e. ∂²W(K₁,K₂,K₃)/∂Kᵢ∂ξⱼ) and kinTerms[i,j,k] (this is equivalent to d^3W/dK_i/dXi_j/dK_kk) =#
+
+function sensitivityFieldFncs(matlModel::Symbol)
+    @variables α K[1:3] # K[1:3] are the orthogonal strain invariants
```

Comment:
Let's discuss if use of `Symbolics` is really required here. I wonder if `ForwardDiff` could be used to get the derivatives numerically instead of symbolically.